pkg/segment/test/segment_test.go

```diff
@@ -123,12 +197,56 @@ func TestRemoveRandom(t *testing.T) {
 			t.Errorf("Iteration %d: failed to find segment with key %d", i, j)
 			break
 		}
+		temprange := seg.Range()
+		s.Remove(seg)
+		nrRemovals++
+		if err := checkSet(&s, testSize-nrRemovals); err != nil {
+			t.Errorf("Iteration %d: %v", i, err)
+			break
+		}
+		if err := s.checkMaxGap(); err != nil {
+			t.Errorf("When removing %v: %v", temprange, err)
+			break
+		}
+	}
+	if got, want := countSegmentsIn(&s), testSize-nrRemovals; got != want {
```

Comment:
These lines seem to be added in every test. I don't object, but you can just make them a common sanityCheck. Eg..

func sanityCheck(t *testing.T, s *Set, segments int) {
   t.Helper()
   if got, want := .... 
   if t.Failed() {
     ....
   }
}

Reply:
This is part of the original test. I'm just reusing them.